api/src/main/java/org/openmrs/obs/handler/BinaryDataHandler.java

```diff
@@ -24,128 +24,130 @@
 import org.springframework.util.Assert;
 
 /**
- * Handler for storing files for complex obs to the file system. Files are stored in the location
- * specified by the global property: "obs.complex_obs_dir"
+ * Handler for storing files for complex obs to the file system. Files are
+ * stored in the location specified by the global property:
+ * "obs.complex_obs_dir"
  * 
  * @since 1.5
  */
 public class BinaryDataHandler extends AbstractHandler implements ComplexObsHandler {
-	
+
 	/** Views supported by this handler */
 	private static final String[] supportedViews = { ComplexObsHandler.RAW_VIEW, };
-	
+
 	private static final Logger log = LoggerFactory.getLogger(BinaryDataHandler.class);
-	
+
 	/**
-	 * Constructor initializes formats for alternative file names to protect from unintentionally
-	 * overwriting existing files.
+	 * Constructor initializes formats for alternative file names to protect from
+	 * unintentionally overwriting existing files.
 	 */
 	public BinaryDataHandler() {
 		super();
 	}
-	
+
 	/**
-	 * Currently supports the following views: org.openmrs.obs.ComplexObsHandler#RAW_VIEW
+	 * Currently supports the following views:
+	 * org.openmrs.obs.ComplexObsHandler#RAW_VIEW
 	 * 
-	 * @see org.openmrs.obs.ComplexObsHandler#getObs(org.openmrs.Obs, java.lang.String)
+	 * @see org.openmrs.obs.ComplexObsHandler#getObs(org.openmrs.Obs,
+	 *      java.lang.String)
 	 */
 	@Override
 	public Obs getObs(Obs obs, String view) {
 		File file = getComplexDataFile(obs);
 		log.debug("value complex: " + obs.getValueComplex());
 		log.debug("file path: " + file.getAbsolutePath());
 		ComplexData complexData = null;
-		
+
 		// Raw view (i.e. the file as is)
 		if (ComplexObsHandler.RAW_VIEW.equals(view)) {
-			// to handle problem with downloading/saving files with blank spaces or commas in their names
+			// to handle problem with downloading/saving files with blank spaces or commas
+			// in their names
 			// also need to remove the "file" text appended to the end of the file name
 			String[] names = obs.getValueComplex().split("\\|");
```

Comment:
If this is still untouched, then the NPE will keep happening.

Please set up first a failing unit test that triggers the bug, and work your way to stop making it fail.

Reply:
Use:
```java
if (!StringUtils.isBlank(obs.getValueComplex())) {
```